src/Integration/Binding/SolutionBindingOperation.cs

```diff
@@ -0,0 +1,204 @@
+//-----------------------------------------------------------------------
+// <copyright file="SolutionBindingOperation.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using EnvDTE;
+using Microsoft.VisualStudio.CodeAnalysis.RuleSets;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+
+namespace SonarLint.VisualStudio.Integration.Binding
+{
+    /// <summary>
+    /// Solution level binding by delegating some of the work to <see cref="ProjectBindingOperation"/>
+    /// </summary>
+    internal class SolutionBindingOperation : IBindingOperation, ISolutionRuleStore

```

Comment:
And if it's the case that the two interfaces _can't_ be implemented by different object because they share state, then why are they different interfaces?